Xamarin.Forms.Core/ResourceDictionary.cs

```diff
@@ -94,6 +147,10 @@ public bool ContainsKey(string key)
 					return _innerDictionary[index];
 				if (_mergedInstance != null && _mergedInstance.ContainsKey(index))
 					return _mergedInstance[index];
+				if (MergedDictionaries != null)
+					foreach (var dict in MergedDictionaries.Reverse())
```

Comment:
Using a for loop on an ICollection, would result in using ElementAt, which is less efficient than just using a foreach loop.